### PR TITLE
Error message when fails to fetch policies

### PR DESCRIPTION
**Pull request by didierofrivia**
**What this PR does / why we need it**:

Renders an error message when it fails to fetch the APIcast policy registry

It currently looks like this:
Screenshot 2019-05-30 at 15 22 00
Screenshot 2019-05-30 at 15 27 55

**Which issue(s) this PR fixes** 

fixes https://issues.jboss.org/browse/THREESCALE-2312

**Verification steps** 

Easy way to test this is in dev mode, start a rails server with `APICAST_REGISTRY_URL` env with nothing or a random URL. Then visit the Service integration page, under policies you should see an error message.

**Special notes for your reviewer**:
TO DO:
* [x] Better Styling cc: @thomasmaas 
* [x] Tests

### CODE REVIEW

File: app/services/policies/policies_list_service.rb [hunk 1]
```diff
@@ -29,18 +35,30 @@ def initialize(account, builtin: true)
...
+    list.merge! fetch_policies_from_apicast if builtin?
+    list.merge! policies_from_account
+    list.to_h
+  end
+
   def fetch_policies_from_apicast
-    response = ::JSONClient.get(apicast_registry_url)
+    begin
+      response = ::JSONClient.get(apicast_registry_url)
+    rescue StandardError => error
```
Comment (guicassolato) [inline app/services/policies/policies_list_service.rb]: Can we try something more specific than `StandardError` here?

**Comment by thomasmaas** [comment]
- As the policies themselves already sit in a box, I'm not convinced another box is needed. If you really want it the borders should not touch at the right.
- Make the second line the header so the actual error is large type (get rid of "Error").

File: test/test_helpers/proxy.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+def stub_apicast
```
Comment (guicassolato) [inline test/test_helpers/proxy.rb]: Can we call it `stub_apicast_registry`?

File: test/test_helpers/proxy.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  url = 'http://apicast.alaska/policies'
+  ThreeScale.config.sandbox_proxy.stubs(apicast_registry_url: url)
+
+  stub_request(:get, url)
+      .with(headers: { 'Accept' => '*/*' })
+      .to_return(status: 200, body: "{\"policies\":{\"apicast\":[{\"description\":\"Main functionality of APIcast.\",\"$schema\":\"http:\\/\\/apicast.io\\/policy-v1\\/schema#manifest#\",\"name\":\"APIcast policy\",\"configuration\":{\"properties\":{},\"type\":\"object\"},\"version\":\"builtin\"}]}}",
```
Comment (guicassolato) [inline test/test_helpers/proxy.rb]: I think you could use here

```ruby
{ policies: { apicast: [JSON.parse(file_fixture('policies/apicast-policy.json').read)] } }.to_json
```

File: app/controllers/api/integrations_controller.rb [hunk 4]
```diff
@@ -13,6 +13,11 @@ class Api::IntegrationsController < Api::BaseController
   rescue_from ActiveRecord::StaleObjectError, with: :edit_stale
 
   def edit
+    begin
+      @registry_policies = Policies::PoliciesListService.call!(current_account)
+    rescue RuntimeError, Errno::ECONNREFUSED => error
```
Comment (hallelujah) [inline app/controllers/api/integrations_controller.rb]: You can use `rescue_from`

File: app/services/policies/policies_list_service.rb [hunk 5]
```diff
@@ -29,18 +35,30 @@ def initialize(account, builtin: true)
...
-    list.merge! fetch_policies_from_apicast if builtin?
-    list.merge! policies_from_account
-    list.to_h
+    policies
   rescue *HTTP_ERRORS, PoliciesListServiceError => error
     Rails.logger.error { error } and return
   end
 
+  def call!
+    policies
```
Comment (hallelujah) [inline app/services/policies/policies_list_service.rb]: I'd say this is not necessarily its own method.

```ruby

  def call
    call!
  rescue *HTTP_ERRORS, PoliciesListServiceError => error
     Rails.logger.error { error } and return
  end

  def call
    list = PolicyList.new
    list.merge! fetch_policies_from_apicast if builtin?
    list.merge! policies_from_account
    list.to_h
  end
```

File: app/views/api/integrations/apicast/shared/_policies.html.slim [hunk 6]
```diff
@@ -1,13 +1,18 @@
...
-    var chain = #{json @proxy.policies_config}
-    var serviceId = #{json @service.id}
-    initPolicies({element: 'policies', registry: registry, chain: chain, serviceId: serviceId})
-  })
+  - if @error.nil?
+    - content_for :javascripts do
+      = javascript_pack_tag 'policies'
+
+    javascript:
+      document.addEventListener('DOMContentLoaded', function () {
```
Comment (hallelujah) [inline app/views/api/integrations/apicast/shared/_policies.html.slim]: So I thought we query the policies through an AJAX request ... I think we should but probably out of the scope of this PR

@3scale/system some wdyt?

File: app/controllers/api/integrations_controller.rb (hunk 4 above)
Comment (Martouta) [reply app/controllers/api/integrations_controller.rb]: I like what Hery is suggesting but that changes a lot the scope of the task. So we should probably create another Jira issue for doing that

File: app/views/api/integrations/apicast/shared/_policies.html.slim (hunk 6 above)
Comment (Martouta) [reply app/views/api/integrations/apicast/shared/_policies.html.slim]: I agree with Hery but I think it should be out of the scope of the PR. And eveb better if it is done after https://github.com/3scale/porta/pull/826/files#discussion_r289074548

**Comment by didierofrivia** [comment]
@hallelujah @Martouta Yeah, I agree with you guys the design was not the best, and it could be improved with caching and/or async with JS. I'll open an issue for the desired improvement. Thanks!

**Comment by thomasmaas** [comment]
Hmm, i only now see the color is no longer red. I wonder if it's clear enough that there is an error.

**Pull request merged**